material/parser.py

```diff
@@ -1,12 +1,41 @@
-from dataclasses import dataclass, field
+import copy
+from dataclasses import dataclass
+import dataclasses
+import functools
 import bpy
+import mathutils
 import numpy as np
-import gpu
-import time
 
-from .tile import get_tile_conf
-from .cc import get_cc_settings
+from .tile import get_tile_conf, F64Texture
+from .cc import SOLID_CC, get_cc_settings
 from .blender import get_blender_settings
+from ..globals import F64_GLOBALS
+
+@functools.cache
+def quantize(x: float, bits: int, mi=0.0, ma=1.0): # quantize in a range
+  value_count = 2**bits-1
+  range_size = ma - mi
+  return mi + round((x - mi) / range_size * value_count) / value_count * range_size
+
+@functools.cache
+def quantize_tuple_cached(t: tuple, bits: int, mi=0.0, ma=1.0):
+  return tuple(quantize(x, bits, mi, ma) for x in t)
+
+def quantize_tuple(t, bits: int, mi=0.0, ma=1.0):
+  return quantize_tuple_cached(tuple(t), bits, mi, ma)
+
+@functools.cache
+def quantize_srgb_cached(linear_color: tuple, include_alpha: bool, force_alpha: bool):
+  result = list(mathutils.Color(linear_color[:3]).from_scene_linear_to_srgb())
+  if include_alpha or force_alpha:
+    if len(linear_color) > 3 and not force_alpha:
+      result.append(linear_color[3])
+    else:
+      result.append(1.0)
```

Comment:
```suggestion
  if force_alpha:
    result.append(1.0)
  elif include_alpha:
    if len(linear_color) > 3:
      result.append(linear_color[3])
    else:
      result.append(1.0)
```
is less confusing to me to read?